sqlalchemy_dremio/base.py

```diff
@@ -0,0 +1,266 @@
+# the MIT License: http://www.opensource.org/licenses/mit-license.php
+
+"""
+Support for the Dremio database.
+
+
+"""
+from sqlalchemy import sql, schema, types, exc, pool
+from sqlalchemy.sql import compiler, expression
+from sqlalchemy.engine import default, base, reflection
+from sqlalchemy import processors
+
+class AcNumeric(types.Numeric):
+    def get_col_spec(self):
+        return "NUMERIC"
+
+    def bind_processor(self, dialect):
+        return processors.to_str
+
+    def result_processor(self, dialect, coltype):
+        return None
+
+class AcFloat(types.Float):
+    def get_col_spec(self):
+        return "FLOAT"
+
+    def bind_processor(self, dialect):
+        """By converting to string, we can use Decimal types round-trip."""
+        return processors.to_str
+
+class AcInteger(types.Integer):
+    def get_col_spec(self):
+        return "INTEGER"
+
+class AcTinyInteger(types.Integer):
+    def get_col_spec(self):
+        return "TINYINT"
+
+class AcSmallInteger(types.SmallInteger):
+    def get_col_spec(self):
+        return "SMALLINT"
+
+class AcDateTime(types.DateTime):
+    def get_col_spec(self):
+        return "DATETIME"
+
+class AcDate(types.Date):
+
+    def get_col_spec(self):
+        return "DATETIME"
+
+class AcText(types.Text):
+    def get_col_spec(self):
+        return "MEMO"
+
+class AcString(types.String):
+    def get_col_spec(self):
+        return "TEXT" + (self.length and ("(%d)" % self.length) or "")
+
+class AcUnicode(types.Unicode):
+    def get_col_spec(self):
+        return "TEXT" + (self.length and ("(%d)" % self.length) or "")
+
+    def bind_processor(self, dialect):
+        return None
+
+    def result_processor(self, dialect, coltype):
+        return None
+
+class AcChar(types.CHAR):
+    def get_col_spec(self):
+        return "TEXT" + (self.length and ("(%d)" % self.length) or "")
+
+class AcBinary(types.LargeBinary):
+    def get_col_spec(self):
+        return "BINARY"
+
+class AcBoolean(types.Boolean):
+    def get_col_spec(self):
+        return "YESNO"
+
+class AcTimeStamp(types.TIMESTAMP):
+    def get_col_spec(self):
+        return "TIMESTAMP"
+
+class DremioExecutionContext(default.DefaultExecutionContext):
+
+    def get_lastrowid(self):
+        self.cursor.execute("SELECT @@identity AS lastrowid")
+        return self.cursor.fetchone()[0]
+
+
+class DremioCompiler(compiler.SQLCompiler):
+    extract_map = compiler.SQLCompiler.extract_map.copy()
+    extract_map.update({
+            'month': 'm',
+            'day': 'd',
+            'year': 'yyyy',
+            'second': 's',
+            'hour': 'h',
+            'doy': 'y',
+            'minute': 'n',
+            'quarter': 'q',
+            'dow': 'w',
+            'week': 'ww'
+    })
+
+    def visit_cast(self, cast, **kwargs):
+        return cast.clause._compiler_dispatch(self, **kwargs)
+
+    def visit_select_precolumns(self, select):
+        """Dremio puts TOP, it's version of LIMIT here """
```

Comment:
Seems like search and replace was used to clobber Access and plop in Dremio.
Dremio does not use TOP.
This leads to some terribly misleading comments.
Lets fix these before they bite us.